src/resmom/linux/alps.c

```diff
@@ -3187,26 +3189,42 @@ inventory_to_vnodes(basil_response_t *brp)
 			default:
 				continue;
 		}
-		for (seg=node->segments; seg; seg=seg->next) {
-			char	vname[128];
 
-			snprintf(vname, sizeof(vname), "%s_%ld_%d",
-				mpphost, node->node_id, seg->ordinal);
-			vname[sizeof(vname)-1]='\0';
+		totcpus = 0;
+		totmem = 0;
```

Comment:
You might want to move the initialization of totcpus and totmem just before the for loop. Also, add a comment indicating these values must be initialized to accomodate !vnode_per_numa_node.